src/operator/tensor/matrix_op-inl.h

```diff
@@ -397,9 +397,7 @@ inline bool SliceForwardInferStorageType(const nnvm::NodeAttrs& attrs,
   const auto& in_stype = in_attrs->at(0);
   auto& out_stype = out_attrs->at(0);
   bool dispatched = false;
-  const bool invalid_ctx = dev_mask != mshadow::cpu::kDevMask;
-  const auto dispatch_ex = invalid_ctx ? DispatchMode::kFComputeFallback :
-                                         DispatchMode::kFComputeEx;
+  const auto dispatch_ex = DispatchMode::kFComputeEx;
```

Comment:
Is the slice test modified appropriately and added to the GPU test suite?

Reply:
`test_sparse_nd_slice` is already import in `tests/python/gpu/test_operator_gpu.py`. `rand_sparse_ndarray` will use default context.